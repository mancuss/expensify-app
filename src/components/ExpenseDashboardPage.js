import React from 'react';
import ExpenseList from './ExpenseList'
import ExpenseListFilters from './ExpenseListFilters'
import ExpesneSummary from './ExpensesSummary'


const ExpenseDashboardPage = () => (
    <div>
        <ExpesneSummary />
        <ExpenseListFilters />
        <ExpenseList />
    </div>
);

export default ExpenseDashboardPage