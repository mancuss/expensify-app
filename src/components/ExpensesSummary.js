import React from 'react'
import { connect } from 'react-redux'
import selectExpenses from '../selectors/expenses'
import selectExpensesTotal from '../selectors/expenses-total'
import numeral from 'numeral'

export const ExpensesSummary = (props) => {
    console.log("Expense Sumamry is here")
    return (
        <div>
           <h1>Viewing {props.expensesCount} {props.expensesCount === 1 ? "expense" : "expenses"} totalling {numeral(props.expensesTotal / 100).format('$0,0.00')}</h1>
        </div>
    )
}

const mapStateToProps = (state) => {
    const expesnes = selectExpenses(state.expenses, state.filters)
    return {
        expensesCount: expesnes.length,
        expensesTotal: selectExpensesTotal(expesnes)
    }

}

export default connect(mapStateToProps)(ExpensesSummary)