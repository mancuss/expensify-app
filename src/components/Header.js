import React from "react";
import { NavLink, Link } from "react-router-dom";
import { connect } from "react-redux";
import { startLogout, startLogin } from "../actions/auth";

export const Header = ({ startLogout }) => (
  <header className='header'>
    <div className='content-container'>
      <div className='header__content'>
        <Link className='header__title' to='/dashboard'>
          <h1>Expensify</h1>
        </Link>

        <button className='button button--link' onClick={startLogout}>
          Logout
        </button>
      </div>
    </div>
  </header>
);

const mapDispatchToProps = (dispatch) => ({
  startLogout: () => dispatch(startLogout()),
});

export default connect(undefined, mapDispatchToProps)(Header);

// export const Header = ({ startLogout }) => (
//   <header>
//     <h1>Expensify</h1>
//     <NavLink to='/dashboard' activeClassName='is-active'>
//       Expensify
//     </NavLink>
//     <NavLink to='/create' activeClassName='is-active'>
//       Create Expense
//     </NavLink>
//     <button onClick={startLogout}>Logout</button>
//   </header>
// );
