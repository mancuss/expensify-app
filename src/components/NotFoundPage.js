import React from 'react';
import { Link } from 'react-router-dom'

const NotFoundPage = () => (
    <div>
        404 - Go to <Link to='/'> HOME </Link>
    </div>
)

export default NotFoundPage