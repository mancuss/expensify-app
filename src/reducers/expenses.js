const expensesReducerDefaultState = [];

export default (state = expensesReducerDefaultState, action) => {
  switch (action.type) {
    case "ADD_EXPENSE":
      return [...state, action.expense];
    case "REMOVE_EXPENSE":
      return state.filter((expense) => expense.id !== action.id);
    case "EDIT_EXPENSE":
      return state.map((expense) => {
        if (expense.id === action.id) {
          //find the corrent expense
          return {
            ...expense, //spread it
            ...action.updates, //spread updates, so new values will override the values from "above"
          };
        } else {
          return expense;
        }
      });
    case "SET_EXPENSES":
      return action.expenses;
    default:
      return state;
  }
};
