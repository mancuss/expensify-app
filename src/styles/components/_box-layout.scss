.box-layout {
    align-items: center; //takes it to the center of the screen vertically 
    background: url('/images/bg.jpg');
    background-size: cover;
    display: flex;
    justify-content: center; //takes it to the center of the screen horizontally 
    height: 100vh;
    width: 100vw;
}

.box-layout__box {
    background: fade-out($color: white, $amount: .15);
    border-radius: 3px;
    padding: $l-size $m-size; //first one is top and botton, second one is left and right
    text-align: center;
    width: 25rem;
}

.box-layout__title {
    margin: 0 0 $m-size 0;
    line-height: 1;
}