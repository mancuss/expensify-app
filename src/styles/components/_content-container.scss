.content-container {
    margin: 0 auto; // auto takes care of showing the container in the middle of the page.
    padding: 0 $m-size;
    max-width: 90rem;

}