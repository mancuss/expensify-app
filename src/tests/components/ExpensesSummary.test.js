import React from "react";
import { shallow } from "enzyme";
import expenses from "../fixtures/expenses";
import { ExpensesSummary } from "../../components/ExpensesSummary";

test("should render Expense Summary component with expenses", () => {
  const wrapper = shallow(<ExpensesSummary expenses={expenses} />);
  expect(wrapper).toMatchSnapshot();
});

test("should render Expense Summary component with one expens", () => {
  const wrapper = shallow(<ExpensesSummary expenses={[expenses[0]]} />);
  expect(wrapper).toMatchSnapshot();
});

test("should render Expense Summary component with empty expense list", () => {
  const wrapper = shallow(<ExpensesSummary expenses={[]} />);
  expect(wrapper).toMatchSnapshot();
});
