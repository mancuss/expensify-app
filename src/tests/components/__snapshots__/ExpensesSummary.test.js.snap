// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Expense Summary component with empty expense list 1`] = `
<div>
  <h1>
    Viewing 
     
    expenses
     totalling 
    $0.00
  </h1>
</div>
`;

exports[`should render Expense Summary component with expenses 1`] = `
<div>
  <h1>
    Viewing 
     
    expenses
     totalling 
    $0.00
  </h1>
</div>
`;

exports[`should render Expense Summary component with one expens 1`] = `
<div>
  <h1>
    Viewing 
     
    expenses
     totalling 
    $0.00
  </h1>
</div>
`;
